src/joint_trajectory_action/minjerk.py

```diff
@@ -0,0 +1,267 @@
+
+    params:
+        m_coeffs: N-dimensional array of (6+1) x k  coefficients
+            for every control point
+            numpy.array of size N by (6 + 1) by k
+            N is the number of control points
+            k is the number of dimensions for each point
+        num_intervals: the number of intervals between
+            control points
+            int > 0
+        duration_array: array of user-supplied control duration of segment
+            numpy.array of size N-1
+            N is the number of control points
+
+    returns:
+        m_curve: positions along the minimum trajectory  in k-dimensions
+            numpy.array of size N*num_interval+1  by k
+            (the +1 is to include the start position on the curve)
+    """
+    assert num_intervals > 0,\
+        "Invalid number of intervals chosen (must be greater than 0)"
+    interval = 1.0 / num_intervals
+    (num_axes, num_mpts, _) = np.shape(m_coeffs)
+    m_curve = np.zeros((num_mpts*num_intervals+1, num_axes))
+    # Copy out initial point
+    m_curve[0, :] = m_coeffs[:, 0, 0]
+    if duration_array == None:
+         duration_array = np.array([1.0]*num_mpts)
+    assert len(duration_array) == num_mpts,\
+         "Invalid number of intervals chosen (must be equal to N={})".format(num_mpts)
+    for current_mpt in range(num_mpts):
+         m_coeff_set = m_coeffs[:, current_mpt, range(7)]
+         for iteration, t in enumerate(np.linspace(interval, 1,
+                                                   num_intervals)):
+              m_curve[(current_mpt *
+                       num_intervals +
+                       iteration+1), :] = _minjerk_trajectory_point(m_coeff_set, t * duration_array[current_mpt])
+    return m_curve
+    
+def _minjerk_trajectory_point(m_coeff, t):
+    """
+    Internal convenience function for calculating
+    a k-dimensional point defined by the supplied
+    minimum jerk coefficients. Finds the point that
+    describes the current position along the minimum
+    trajectory segment for k dimensions.
+
+    params:
+        m_coeff => m0...m3: Four k-dimensional minimum jerk
+            coefficients each one is a numpy.array
+            of size k by 1, so
+            m_coeff is a numpy array of size k by (6+1)
+            k is the number of dimensions for each
+            coefficient
+        t: percentage of time elapsed for this segment
+            0 <= int <= 1.0
+
+    returns:
+        current position in k dimensions
+            numpy.array of size 1 by k
+    """
+    a0 = m_coeff[:,0]
+    a1 = m_coeff[:,1]
+    a2 = m_coeff[:,2]
+    a3 = m_coeff[:,3]
+    a4 = m_coeff[:,4]
+    a5 = m_coeff[:,5]
+    tm = m_coeff[:,6]
+
+    t = t * tm # input t is percentage of time elapsed for this segment, tm is the duration of this segment and to calculate x, v, a , t is the time[s] elapsed for this segment
+
+    # calculate x, v, z at the time percentage  t
+    x=a0+a1*t+a2*t*t+a3*t*t*t+a4*t*t*t*t+a5*t*t*t*t*t;
```

Comment:
Since you are already using `numpy`, I think it makes sense to leverage `numpy.power()` for readability in these equations, eg.
```
x=a0+a1*t+a2*np.power(t,2)+a3*np.power(t,3)+a4*np.power(t,4)+a5*np.power(t,5);
```